service/prometheus/service.go

```diff
@@ -0,0 +1,81 @@
+package prometheus
+
+import (
+	"net/http"
+	"net/url"
+	"path"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+)
+
+const (
+	prometheusReloadPath = "/-/reload"
+)
+
+type Prometheus interface {
+	Reload() error
+}
+
+type Config struct {
+	Logger micrologger.Logger
+
+	PrometheusAddress string
+}
+
+func DefaultConfig() Config {
+	return Config{
+		Logger: nil,
+
+		PrometheusAddress: "",
+	}
+}
+
+type Service struct {
+	logger micrologger.Logger
+
+	prometheusAddress string
+}
+
+func New(config Config) (*Service, error) {
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.Logger must not be empty")
+	}
+
+	if config.PrometheusAddress == "" {
+		return nil, microerror.Maskf(invalidConfigError, "config.PrometheusAddress must not be empty")
+	}
+
+	u, err := url.ParseRequestURI(config.PrometheusAddress)
```

Comment:
Doing this to both validate the URL, and also append the path. Flags just need to specify the host+port towards prometheus.